### PR TITLE
Feature 8323 - GETINFO md/all

### PR DESCRIPTION
**Pull request by rl1987**
https://trac.torproject.org/projects/tor/ticket/8323

### CODE REVIEW

File: src/or/nodelist.c [hunk 1]
```diff
@@ -891,6 +891,15 @@ node_get_by_hex_id(const char *hex_id, unsigned flags)
   return NULL;
 }
 
+const smartlist_t *
```
Comment (asn-d6) [inline src/or/nodelist.c]: Hmm why do we need to define a new function for this? We already have `nodelist_get_list()` that does this.
See how it's used in other parts of the code. :)

File: src/or/control.c [hunk 2]
```diff
@@ -2207,6 +2207,25 @@ getinfo_helper_dir(control_connection_t *control_conn,
         return -1;
       }
     }
+  } else if (!strcmp(question, "md/all")) {
+    const smartlist_t *nodes = nodelist_get_all_nodes();
+    if (!nodes) {
```
Comment (asn-d6) [inline src/or/control.c]: Hmm, I don't think we should return `Internal error` if there are no nodes/consensus. Instead we should do what we do for `ns/all` in `getinfo_helper_networkstatus()` and return an empty string.

That's also an artficat of the way your `nodelist_get_all_nodes()` function works. If you use the `nodelist_get_list()` function, you are not gonna have this problem.

Comment (rl1987) [reply src/or/control.c]: Fixed in 51641533b2c565e22e24b27bf5782d70802a051.

File: src/or/nodelist.c (hunk 1 above)
Comment (rl1987) [reply src/or/nodelist.c]: Fixed.

**Pull request closed without merging**